conan/tools/cmake/presets.py

```diff
@@ -133,14 +133,19 @@ def write_cmake_presets(conanfile, toolchain_file, generator, cache_variables):
         if "CMAKE_SH" not in cache_variables:
             cache_variables["CMAKE_SH"] = "CMAKE_SH-NOTFOUND"
 
-        cmake_make_program = conanfile.conf.get("tools.gnu:make_program", default=cache_variables.get("CMAKE_MAKE_PROGRAM"))
+        cmake_make_program = conanfile.conf.get("tools.gnu:make_program",
+                                                default=cache_variables.get("CMAKE_MAKE_PROGRAM"))
         if cmake_make_program:
             cmake_make_program = cmake_make_program.replace("\\", "/")
             cache_variables["CMAKE_MAKE_PROGRAM"] = cmake_make_program
 
     if "CMAKE_POLICY_DEFAULT_CMP0091" not in cache_variables:
         cache_variables["CMAKE_POLICY_DEFAULT_CMP0091"] = "NEW"
 
+    if "BUILD_TESTING" not in cache_variables:
```

Comment:
What about `CMakeToolchain.variables`? Is it a possible scenario, right?

Reply:
It sounds like ``CMakeToolchain.variables`` should be for user variables, while everything that is a CMake built-in, should be to ``cache_variables`` otherwise there is a risk that the ``variables`` will not have enough priority. @jcar87 wdyt?

Reply:
I'd say that's a good approximation!

I can find widespread evidence on GitHub that `BUILD_TESTING` is treated as a cache variable by projects:
1) For example [here](https://github.com/dduan/Pathos/blob/5efb5fdfd6d89cd394c0f9e3094336628d8d9903/CMakeLists.txt#L8-L9) by calling `option()` 

2) It will also be a cache variable if  the user provides it via `-DBUILD_TESTING=ON/OFF` - in this case, the user's preference will take precedence and be respected regardless of whether we define it as a `cache_variable` or the project sets an `option()` - 

So I think the way it currently is in this PR is the safest and least likely to have side effects - especially if there's any chance projects define or reference `BUILD_TESTING` before the first call to `project()` (have found  examples [here](https://github.com/johannes51/AerospaceFramework/blob/1ab876ce80ea67a305ba0fe03eb9ab172d1ff24c/dependencies/CMakeLists.txt#L3) and [here](https://github.com/EagleDad/cmake-playground/blob/c80eebec3c667e53e47f96eb89da30dc748a4980/CMakeLists.txt#L22))

Reply:
Ok, lets recommend that in the docs then 👍

Reply:
I'm not in the opposite of using `cached_variables`, but still, users are free to use both and it will have a different behavior, depending the usage.

Reply:
Yes, but if we explicitly say it in the docs, they should use each one for the documented cases, so they will not be "free" to use them invariantly for all cases.